tensorflow/core/grappler/optimizers/arithmetic_optimizer.cc

```diff
@@ -3625,6 +3740,8 @@ Status ArithmeticOptimizer::SimplifyArithmeticOps(bool can_use_shapes) {
     pipeline.AddStage<RemoveStackStridedSliceSameAxis>(ctx, ctx_ext);
   if (options_.fuse_squared_diff)
     pipeline.AddStage<FuseSquaredDiffStage>(ctx, ctx_ext);
+  if (options_.fuse_mul_add)
+    pipeline.AddStage<FuseMulAddStage>(ctx, ctx_ext);
```

Comment:
@jerryyin per discussion let's try remove this stage should XLA is enabled.

Reply:
Thanks this is exactly where I'm heading.

Reply:
try check if `OptimizerOptions` can be accessed here. in XLA passes such as `compiler/jit/mark_for_compilation_pass.cc` that's the structure be inspected to understand XLA JIT levels.

Reply:
I can't find it easily accessible unless we change its constructor together with the way it initializes. That seems to be impacting a wider range of components. I need to study a bit more about one optimization pass `meta_optimizer` that take advantage of this `OptimizerOptions`.

On the other hand, I'm going to double check if the easier compile-time option is do-able, aka using the `TF_ENABLE_XLA` option to side way it.

Reply:
Disabling it on TF_ENABLE_XLA seems like overkill, it'll disable it in many scenarios where it does not need to be.

Reply:
The other fused ops are inserted only in the non-XLA path. They are inserted in a pass which I believe runs quite late, after even the GPU placement is done for the nodes.

One option would be switch the FMA node creation to that later pass, and thus bypass this matter altogether.

Reply:
see here : https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/782/files?file-filters%5B%5D=.cc#diff-99de50a7ff03ef1c38515c21b3a4a934R258-R264

Reply:
I prototyped a solution that pokes a hole so that the `meta_optimizer` can pass the `XlaGlobalJitLevel` from `xla_config_registry::GetGlobalJitLevel`. I'm surprised to find that both `general` and `single_gpu` option are marked as `OptimizerOptions::OFF`. I was expecting it being marked as either `OptimizerOptions::ON_1` or `OptimizerOptions::ON_2` but turns out it is not the case. I'm suspecting that it is due to this been too early in the optimization stage that XLA as a whole is not turned on yet. (Please refer to 83aec9d2c9c622a867e4dbd139f20ef9a05bf540 for my changes)

Based on this, I believe the only option level to to move this to `layout_optimizer` as @deven-amd suggested. @ekuznetsov139 Can you give that a shot?

Reply:
Thanks for trying. I'll give it a shot.